### PR TITLE
[api-documenter] Support for generating hyperlinks for aliased types

### PR DESCRIPTION
**Pull request by hiranya911**
Adds hyperlinks to referenced types in type aliases.

When a type alias references other types in the same doc model, there's currently no easy way to navigate to those referenced types. With this change, the generated markdown will include links to such referenced types.

Sample output:
Screen Shot 2020-10-12 at 2 13 56 PM

### CODE REVIEW

**Comment by hiranya911** [comment]
@octogonz any thoughts on this change?

File: common/changes/@microsoft/api-documenter/master_2020-10-12-21-25.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/api-documenter",
+      "comment": "Support for generating hyperlinks from type aliases",
+      "type": "patch"
```
Comment (octogonz) [inline common/changes/@microsoft/api-documenter/master_2020-10-12-21-25.json:6]: Make it a "minor" change

File: apps/api-documenter/src/documenters/MarkdownDocumenter.ts [hunk 2]
```diff
@@ -37,7 +37,9 @@ import {
   ApiDeclaredItem,
```
Comment (octogonz) [inline apps/api-documenter/src/documenters/MarkdownDocumenter.ts:37]: Could you add a test case to [build-tests/api-documenter-test](https://github.com/microsoft/rushstack/tree/master/build-tests/api-documenter-test)?  It doesn't need to be exhaustive, just something to show how the alias gets rendered.

File: apps/api-documenter/src/documenters/MarkdownDocumenter.ts [hunk 3]
```diff
@@ -347,6 +349,28 @@ export class MarkdownDocumenter {
         output.appendNode(extendsParagraph);
       }
     }
+
+    if (apiItem instanceof ApiTypeAlias) {
+      const refs: ExcerptToken[] = apiItem.excerptTokens.filter(
```
Comment (octogonz) [inline apps/api-documenter/src/documenters/MarkdownDocumenter.ts:354]: There are a couple interesting edge cases for this:

1. The `build-tests/api-documenter-test/etc/markdown/api-documenter-test.exampletypealias.md` file gets rendered without a hyperlink:

```md
<b>Signature:</b>

```typescript
export declare type ExampleTypeAlias = Promise<boolean>;
```
<b>References:</b> Promise
```
I think this is because `Promise` does have a canonical reference but did not get mapped.  We probably should exclude it in that case.

2. If a type is referenced twice, it does not get deduplicated. For example `type X = Y | typeof Y`.

**Comment by octogonz** [comment]
> any thoughts on this change?

@hiranya911 Sorry it took me a while to get to this. We've been really busy this fall and I have not had much time to keep up with API Extractor.

The underlying problem is that Markdown does not allow hyperlinks in a code block.  Your approach of showing a list of "**References:**" seems like a pretty good solution.

I added some feedback.

File: common/changes/@microsoft/api-documenter/master_2020-10-12-21-25.json (hunk 1 above)
Comment (hiranya911) [reply common/changes/@microsoft/api-documenter/master_2020-10-12-21-25.json]: Done

File: apps/api-documenter/src/documenters/MarkdownDocumenter.ts (hunk 2 above)
Comment (hiranya911) [reply apps/api-documenter/src/documenters/MarkdownDocumenter.ts:37]: Added test case (I think I did it right. I couldn't find any info about how to actually run it locally, so I couldn't verify.)

File: apps/api-documenter/src/documenters/MarkdownDocumenter.ts (hunk 3 above)
Comment (hiranya911) [reply apps/api-documenter/src/documenters/MarkdownDocumenter.ts:354]: 1. Implemented a fix to check if the type actually resolves to another locally declared type before adding a reference.
2. Implemented a simple duplicate checker based on text (please let me know if we should use some other criteria here).

Added test cases for both scenarios.

**Comment by hiranya911** [comment]
Thanks @octogonz. I've made the suggested changes. Please take a look when you got a moment.

**Review by octogonz — APPROVED** [review]

**Comment by octogonz** [comment]
This was released with API Extractor 7.12.0 / API Documenter 7.11.0
